onnxruntime/test/framework/ep_plugin_provider_test.cc

```diff
@@ -317,4 +334,129 @@ TEST(PluginExecutionProviderTest, InferOrtDeviceFromDeviceMemoryInfo) {
 #endif  // !defined(ORT_NO_EXCEPTIONS)
 }
 
+static void LoadModelAndAssignNodesToEp(const ORTCHAR_T* model_path,
+                                        const char* ep_name,
+                                        std::unordered_set<std::string> ep_node_names,
+                                        /*out*/ std::shared_ptr<Model>& model) {
+  ASSERT_STATUS_OK(Model::Load(model_path, model, nullptr,
```

Comment:
note that a failure here will only exit early from this helper function. might want to use ASSERT_NO_FATAL_FAILURE at the call site.

https://google.github.io/googletest/advanced.html#asserting-on-subroutines